flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -1811,6 +1834,12 @@ public void testJobMasterOnlyTerminatesAfterTheSchedulerHasClosed() throws Excep
         schedulerTerminationFuture.complete(null);
 
         jobMasterTerminationFuture.get();
+        assertJobMasterResourceClosed(jobMaster);
+    }
+
+    private void assertJobMasterResourceClosed(JobMaster jobMaster) {
```

Comment:
I think we'd better test it in `RpcEndpointTest` and `FencedRpcEndpointTest`. It would be good to cover the scenario in which the fencing token changed.